### PR TITLE
Simplifying parameter sweep API

### PR DESCRIPTION
**Pull request by bknueven**
## Fixes/Addresses:
@TimBartholomew and @adam-a-a noticed an issue in how the parameter sweep tool handles file names with sub-directories (see #485). This PR side-steps the issue by no longer doing path manipulation and just forcing the user to specify a `csv_results_file_name` and/or a `h5_results_file_name`.

## Changes proposed in this PR:
- Separate out `results_file_name` into `csv_results_file_name` and `h5_results_file_name`.
- Remove `write_h5` and `write_csv` flags

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/tools/parameter_sweep.py [hunk 1]
```diff
@@ -619,7 +617,9 @@ def _write_debug_data(
 
     if write_h5:
         fname_h5 = f"local_results_{rank:03}.h5"
-        _write_output_to_h5(local_results_dict, debugging_data_dir, fname_h5)
+        _write_output_to_h5(
+            local_results_dict, os.path.join(debugging_data_dir, fname_h5)
+        )
     if write_csv:
         fname_csv = f"local_results_{rank:03}.csv"
```
Comment (adam-a-a) [inline watertap/tools/parameter_sweep.py:624]: Just doing a quick skim, it seems like you removed `write_h5` and `write_csv` arguments, but not sure that you meant to keep them in here.

Comment (adam-a-a) [reply watertap/tools/parameter_sweep.py:624]: OK- I see what you did here. 

`            h5_results_file_name is not None,
            csv_results_file_name is not None` provide the boolean vals for these in this particular function.

Comment (bknueven) [reply watertap/tools/parameter_sweep.py:624]: Yes -- and in this function it's their existence, not their names, that matter.

**Review by adam-a-a — APPROVED** [review]
LGTM

**Review by eyoung55 — APPROVED** [review]
Looking good, thanks Ben!

**Pull request merged**